src/main/java/com/myMoneyTracker/controller/ExpenseController.java

```diff
@@ -94,7 +95,29 @@
         }
         return new ResponseEntity<List<ExpenseDTO>>(createExpenseDTOs(expenses), HttpStatus.OK);
     }
-    
+
+    @RequestMapping(value = "/find/{category_name:.+}/{currency}/{start_time_millis}/{end_time_millis}", method = RequestMethod.GET)
+    public ResponseEntity<List<ExpenseDTO>> listAllExpensesByCategoryNameAndTimeInterval(
+            @PathVariable("category_name") String categoryName,
+            @PathVariable("currency") String currency,
+            @PathVariable("start_time_millis") long startTimeMillis,
+            @PathVariable("end_time_millis") long endTimeMillis) {
+
```

Comment:
Add a validation clause for currency. If Currency code is not valid throw BadRequestException